src/js_native_api.h

```diff
@@ -111,6 +111,10 @@ NAPI_EXTERN napi_status napi_create_range_error(napi_env env,
                                                 napi_value code,
                                                 napi_value msg,
                                                 napi_value* result);
+NAPI_EXTERN napi_status napi_create_syntax_error(napi_env env,
```

Comment:
This is a new api and it should be enabled / disabled throw `NAPI_EXPERIMENTAL`:
```cpp
#ifdef NAPI_EXPERIMENTAL
NAPI_EXTERN 
napi_status napi_create_syntax_error(napi_env env,
                                     napi_value code,
                                     napi_value msg,
                                     napi_value* result);
#endif  // NAPI_EXPERIMENTAL
```